android-sdk/src/main/java/ai/evolv/android_sdk/EvolvClientImpl.java

```diff
@@ -194,50 +178,228 @@ public void invoke(Object object) {
+                JsonElement allocations = JsonNull.INSTANCE;
+                if (remoteContext.has("experiments")) {
+                    allocations = remoteContext.get("experiments");
+                }
+
+                if (allocations.equals(JsonNull.INSTANCE)
+                        || evolvStore.config.size() == 0
+                        || allocations.getAsJsonObject().size() == 0) {
+                    return;
+                }
+
+                JsonArray entryPointEids = evolvStore.activeEntryPoints();
+
+                if (entryPointEids.size() == 0) {
+                    return;
+                }
+
+                JsonObject confirmations = new JsonObject();
+                if (remoteContext.has("confirmations")) {
+                    confirmations = remoteContext.get("confirmations").getAsJsonObject();
+                }
+
+                JsonObject confirmedCids = new JsonObject();
+                for (Map.Entry<String, JsonElement> entry : confirmations.entrySet() ) {
+                    confirmedCids = entry.getValue().getAsJsonObject().get("cid").getAsJsonObject();
+                }
+
+                JsonObject contaminations = new JsonObject();
+                if (remoteContext.has("contaminations")) {
+                    confirmations = remoteContext.get("contaminations").getAsJsonObject();
+                }
+
+                JsonObject contaminatedCids = new JsonObject();
+                for (Map.Entry<String, JsonElement> entry : contaminations.entrySet() ) {
+                    confirmedCids = entry.getValue().getAsJsonObject().get("cid").getAsJsonObject();
+                }
+
+                JsonArray confirmableAllocations = new JsonArray();
+                for (Map.Entry<String, JsonElement> alloc : allocations.getAsJsonObject().get("allocations").getAsJsonObject().entrySet()) {
+
+                    String cid = alloc.getValue().getAsJsonObject().get("cid").getAsString();
+                    String eid = alloc.getValue().getAsJsonObject().get("eid").getAsString();
+
+                    if (!confirmedCids.has(cid)
+                            && !contaminatedCids.has(cid)
+                            && hasEntryPointEids(evolvStore.activeEids, eid)) {
+                        confirmableAllocations.add(alloc.getValue());
+                    }
+                }
+
+                if (confirmableAllocations.size() == 0) {
+                    return;
+                }
+
+                long timestamp = (new Date()).getTime();
+                JsonObject contextConfirmations = new JsonObject();
+
+                for (JsonElement alloc : confirmableAllocations) {
+                    contextConfirmations.addProperty("cid", alloc.getAsJsonObject().get("cid").getAsString());
+                    contextConfirmations.addProperty("timestamp", timestamp);
+                }
+
+                contextConfirmations.add("confirmations", confirmations);
```

Comment:
The format of contextConfirmations looks wrong here - wouldn't we overwrite "cid" for every active experiment, so we'd only get one? And then we add the previous confirmation set under "confirmations"

Reply:
"The format of contextConfirmations looks wrong here - wouldn't we overwrite "cid" for every active experiment, so we'd only get one? " - this is what happens, because the "cid" key in the object is the same and this means that each next "cid" key takes the place of the previous one 
"And then we add the previous confirmation set under "confirmations" - I don't quite understand, I will figure out

Reply:
i will fix it
![изображение](https://user-images.githubusercontent.com/28364912/127871771-e9550ab2-e1c3-4f44-bfd5-c75f420215bc.png)

Reply:
A user can be in multiple candidates at the same time

Reply:
fixed